src/coreComponents/physicsSolvers/multiphysics/PoromechanicsInitialization.cpp

```diff
@@ -0,0 +1,134 @@
+};
+
+}
+
+// provide a definition for catalogName()
+template< typename POROMECHANICS_SOLVER >
+string
+PoromechanicsInitialization< POROMECHANICS_SOLVER >::
+catalogName()
+{
+  return PoromechanicsCatalogNames< POROMECHANICS_SOLVER >::name();
+}
+
+template< typename POROMECHANICS_SOLVER >
+PoromechanicsInitialization< POROMECHANICS_SOLVER >::
+PoromechanicsInitialization( const string & name,
+                             Group * const parent ):
+  TaskBase( name, parent ),
+  m_poromechanicsSolverName()
+{
+  enableLogLevelInput();
+
+  registerWrapper( viewKeyStruct::poromechanicsSolverNameString(), &m_poromechanicsSolverName ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Name of the poromechanics solver" );
+
+  registerWrapper( viewKeyStruct::performStressInitializationString(), &m_performStressInitialization ).
+    setApplyDefaultValue( true ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setDescription( "Flag to indicate that the solver is going to perform stress initialization" );
+}
+
+template< typename POROMECHANICS_SOLVER >
+PoromechanicsInitialization< POROMECHANICS_SOLVER >::~PoromechanicsInitialization() {}
+
+template< typename POROMECHANICS_SOLVER >
+void
+PoromechanicsInitialization< POROMECHANICS_SOLVER >::
+postProcessInput()
+{
+  ProblemManager & problemManager = this->getGroupByPath< ProblemManager >( "/Problem" );
+  PhysicsSolverManager & physicsSolverManager = problemManager.getPhysicsSolverManager();
+
+  GEOSX_THROW_IF( !physicsSolverManager.hasGroup( m_poromechanicsSolverName ),
+                  GEOSX_FMT( "Task {}: physics solver named {} not found",
+                             getName(), m_poromechanicsSolverName ),
+                  InputError );
+
+  m_poromechanicsSolver = &physicsSolverManager.getGroup< POROMECHANICS_SOLVER >( m_poromechanicsSolverName );
+}
+
+template< typename POROMECHANICS_SOLVER >
+bool
+PoromechanicsInitialization< POROMECHANICS_SOLVER >::
+execute( real64 const time_n,
+         real64 const GEOSX_UNUSED_PARAM( dt ),
+         integer const GEOSX_UNUSED_PARAM( cycleNumber ),
+         integer const GEOSX_UNUSED_PARAM( eventCounter ),
+         real64 const GEOSX_UNUSED_PARAM( eventProgress ),
+         DomainPartition & GEOSX_UNUSED_PARAM( domain ) )
+{
+  if( m_performStressInitialization )
+  {
+    GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "Task `{}`: at time {}s, physics solver `{}` is set to perform stress initialization during the next time step(s)",
+                                          getName(), time_n, m_poromechanicsSolverName ) );
+    m_poromechanicsSolver->performStressInitialization( true );
+  }
+  else
+  {
+    GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "Task `{}`: at time {}s, physics solver `{}` has completed stress initialization",
+                                          getName(), time_n, m_poromechanicsSolverName ) );
+    m_poromechanicsSolver->performStressInitialization( false );
+  }
+
+  return false;
```

Comment:
So it always returns `false`?

Reply:
I added a comment before the `return` to explain why this function always returns `false`. In fact, all the `execute` functions must return false, because `true` is the signal for early return and terminates the simulation with the following message:
```
***** WARNING
***** LOCATION: /g/g90/hamon1/geosx/GEOSX/src/main/main.cpp:46
***** Controlling expression (should be false): state.getState() != State::COMPLETED
Simulation exited early.
```
See here in the `EventManager`: https://github.com/GEOSX/GEOSX/blob/c5d775303be29bad5956bc303d32da4571a9c095/src/coreComponents/events/EventManager.cpp#L186-L201